src/code/z_collision_check.c

```diff
@@ -0,0 +1,4032 @@
+    }
+    sOCResetFuncs[collider->shape](ctxt, collider);
+
+    if (collider->actor != NULL && collider->actor->update == NULL) {
+        return -1;
+    }
+    if (colCtxt->colOCCount >= ARRAY_COUNT(colCtxt->colOC)) {
+        return -1;
+    }
+    if (colCtxt->sacFlags & SAC_ON) {
+        return -1;
+    }
+    index = colCtxt->colOCCount;
+    colCtxt->colOC[colCtxt->colOCCount++] = collider;
+
+    return index;
+}
+
+/**
+ * Sets collider as an OC (object collider) for the current frame, allowing it to detect other OCs.
+ * If CollisionCheck_SAC is enabled, the collider will be inserted into the list at the specified index, otherwise it
+ * will be inserted into the next slot.
+ */
+s32 CollisionCheck_SetOC_SAC(GlobalContext* ctxt, CollisionCheckContext* colCtxt, Collider* collider, s32 index) {
+    if (func_8016A01C(ctxt)) {
+        return -1;
+    }
+    sOCResetFuncs[collider->shape](ctxt, collider);
+
+    if (collider->actor != NULL && collider->actor->update == NULL) {
+        return -1;
+    }
+    if (colCtxt->sacFlags & SAC_ON) {
+        if (index >= colCtxt->colOCCount) {
+            return -1;
+        }
+        //! @bug should be colOC
+        colCtxt->colAT[index] = collider;
+    } else {
+        if (colCtxt->colOCCount >= ARRAY_COUNT(colCtxt->colOC)) {
+            return -1;
+        }
+        index = colCtxt->colOCCount;
+
+        colCtxt->colOC[colCtxt->colOCCount] = collider;
+        colCtxt->colOCCount++;
+    }
+    return index;
+}
+
+/**
+ * Sets a line as an OC collider for this frame.
+ */
+s32 CollisionCheck_SetOCLine(GlobalContext* ctxt, CollisionCheckContext* colCtxt, OcLine* line) {
+    s32 index;
+
+    if (func_8016A01C(ctxt)) {
+        return -1;
+    }
+
+    Collider_ResetLineOC(ctxt, line);
+
+    if (colCtxt->colLineCount >= ARRAY_COUNT(colCtxt->colLine)) {
+        return -1;
+    }
+    index = colCtxt->colLineCount;
+    colCtxt->colLine[colCtxt->colLineCount++] = line;
+
+    return index;
+}
+
+/**
+ * Skips AT elements that are off.
+ */
+s32 CollisionCheck_SkipTouch(ColliderInfo* info) {
+    if (!(info->toucherFlags & TOUCH_ON)) {
+        return 1;
+    }
+    return 0;
+}
+
+/**
+ * Skips AC elements that are off.
+ */
+s32 CollisionCheck_SkipBump(ColliderInfo* info) {
+    if (!(info->bumperFlags & BUMP_ON)) {
+        return 1;
+    }
+    return 0;
+}
+
+/**
+ * If the AT element has no dmgFlags in common with the AC element, no collision happens.
+ */
+s32 CollisionCheck_NoSharedFlags(ColliderInfo* toucher, ColliderInfo* bumper) {
+    if (!(toucher->toucher.dmgFlags & bumper->bumper.dmgFlags)) {
+        return 1;
+    }
+    return 0;
+}
+
+/**
+ * Spawns no blood drops.
+ * Used by collider types HIT1, HIT3, HIT5, METAL, NONE, WOOD, HARD, and TREE
+ */
+void CollisionCheck_NoBlood(GlobalContext* ctxt, Collider* collider, Vec3f* v) {
+}
+
+/**
+ * Spawns blue blood drops.
+ * Used by collider types HIT0 and HIT8.
+ */
+#ifdef NON_MATCHING
+// needs in-function static bss
```

Comment:
What made you decide to leave the bss as extern? You should be able to migrate the data sections.

Reply:
Since bss is not disassembled I am reluctant to include it in the file. Also, this file suffers from bss reordering issues in OoT and is likely to also suffer from such issues here as well and as headers here are liable to change much more often than OoT it would require constant fixing to maintain OK, so seems better to leave it for now for that reason as well.

Reply:
Yeah, you make good points. We can deal with this later when headers are less in flux and we have split bss. Sounds good to me.